instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/TracingExecutionInterceptor.java

```diff
@@ -35,101 +42,121 @@
   static final ExecutionAttribute<Span> SPAN_ATTRIBUTE =
       new ExecutionAttribute<>("io.opentelemetry.auto.Span");
 
+  static final ExecutionAttribute<RequestType> REQUEST_TYPE_ATTRIBUTE =
+      new ExecutionAttribute<>("io.opentelemetry.auto.aws.RequestType");
+
   static final String COMPONENT_NAME = "java-aws-sdk";
 
-  private final Kind kind;
+  private final SpanFactory spanFactory;
+
+  private final Map<RequestType, SdkRequestDecorator> decoratorsMap = mapDecorators();
+  private final Map<String, String> fieldToAttribute = mapFieldToAttribute();
+
+  private Map<RequestType, SdkRequestDecorator> mapDecorators() {
+    Map<RequestType, SdkRequestDecorator> result = new EnumMap<>(RequestType.class);
+    result.put(RequestType.DynamoDB, new DbRequestDecorator());
+    return result;
```

Comment:
`unmodifiableMap` for both

Reply:
Not sure about that. Maps are private, not exposed to external (not leaking), unmod view will definitely add some CPU cycles (delegating methods). Don't see benefit here (it would be different if the map was exposed or leaking).